src/System.Devices.Gpio/System.Devices.Gpio.Samples/Bme280.cs

```diff
@@ -0,0 +1,419 @@
+            if (_spiDevice != null)
+            {
+                _spiDevice.Dispose();
+                _spiDevice = null;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the temperature calibration offset in degrees celsius.
+        /// </summary>
+        public float TemperatureCalibrationOffset { get; set; }
+
+        public void ReadSensor()
+        {
+            ReadTemperature();
+            ReadHumidity();
+            ReadPressure();
+        }
+
+        public float TemperatureInCelsius => _temperature + TemperatureCalibrationOffset;
+
+        public float TemperatureInFahrenheit => TemperatureInCelsius * 1.8F + 32;
+
+        public float Humidity => _humidity;
+
+        public float PressureInPascals => _pressure;
+
+        public float PressureInHectopascals => _pressure / 100;
+
+        public float PressureInMillibars => PressureInHectopascals;
+
+        public bool Begin()
+        {
+            Dispose();
+
+            _csPin.Mode = PinMode.Output;
+            DigitalWrite(_csPin, PinValue.High);
+
+            if (_spiSettings != null)
+            {
+                // Hardware SPI
+                _spiSettings.Mode = SpiMode.Mode0;
+                _spiSettings.ClockFrequency = 500 * 1000;
+                _spiSettings.DataBitLength = 8;
+
+                _spiDevice = new UnixSpiDevice(_spiSettings);
+            }
+            else
+            {
+                // Software SPI
+                _sckPin.Mode = PinMode.Output;
+                _mosiPin.Mode = PinMode.Output;
+                _misoPin.Mode = PinMode.Input;
+            }
+
+            byte chipId = Read8(BME280_REGISTER_CHIPID);
+
+            if (chipId != 0x60)
+            {
+                Console.WriteLine($"chipId = '{chipId:X2}'");
+                return false;
+            }
+
+            ReadSensorCoefficients();
+
+            // Set Humidity oversampling to 1
+            // Set before CONTROL (DS 5.4.3)
+            Write8(BME280_REGISTER_CONTROLHUMID, 0x01);
+            Write8(BME280_REGISTER_CONTROL, 0x3F);
+            return true;
+        }
+
+        private void ReadTemperature()
+        {
+            int var1, var2;
+
+            uint adc_T = Read24(BME280_REGISTER_TEMPDATA);
+            adc_T >>= 4;
+
+            var1 = (((int)((adc_T >> 3) - (_cal_data.dig_T1 << 1))) *
+                     _cal_data.dig_T2) >> 11;
+
+            var2 = ((((int)((adc_T >> 4) - ((int)_cal_data.dig_T1)) *
+                       (int)((adc_T >> 4) - ((int)_cal_data.dig_T1))) >> 12) *
+                     _cal_data.dig_T3) >> 14;
+
+            _tFine = var1 + var2;
+            _temperature = (_tFine * 5 + 128) >> 8;
+            _temperature = _temperature / 100;
+        }
+
+        private void ReadPressure()
+        {
+            long var1, var2, p;
+
+            uint adc_P = Read24(BME280_REGISTER_PRESSUREDATA);
+            adc_P >>= 4;
+
+            var1 = ((long)_tFine) - 128000;
```

Comment:
as a sample this should have better variable names and comments

Reply:
Idem as above comment.

Those are the names used by the official algorithm in the sensor specification. Not sure what they mean.

https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BME280_DS001-12.pdf